students/ZhuojingXie/lesson09/mailroom5.py

```diff
@@ -0,0 +1,111 @@
+import sys
+import os
+from collections import defaultdict
+donor_data = defaultdict(list,{'Andy':[960,256,123.5,40],'Bryce':[30,45,27],
+                            'Charile': [25,50], 'David':[10], 'Elaine' :[75,26]})
+
+class Donor():
+
+
+    """this is Donor class to save donor's information"""
+    def __init__(self, name, donation):
+        self.name = name
+        self.donation = donation
+
+
+    def add_donor_in_data(self):
+        return donor_data[self.name].append(self.donation)
+
+
+class Collection:
+
+    def __init__(self, donor_data):
+        self.donor_data = donor_data
+
+    def all_donor(self):
+        temp_keys=[]
+        for keys in self.donor_data:
+            temp_keys.append(keys)
+        return temp_keys
+
+    def sum_donation(self,name):
+        return sum(self.donor_data[name])
+
+    def last_donation(self, name):
+        return self.donor_data[name][-1]
+
+    def send_a_thank_you(self):
+        print('\n')
+        print(self.all_donor())
+        donor_name =input('\n\nPlease input the name or the donator > ')
+        donation = input('\n\nThank you for donation , please type '
+                    'the money you want to donate > ')
+        while not donation.isdigit():
+            donation = input('\n\nPlease type number only > ')
```

Comment:
per instructions in the assignment there should be no user input code in the Data handling class.


your CLI flow should be separate and class _only_ handle data collection logic.